src/main/java/de/dhbw/mh/slang/antlr/AstBuilder.java

```diff
@@ -60,6 +65,15 @@ public AstNode visitExponentiation( ExponentiationContext ctx ){
 	
 	@Override
 	public AstNode visitAtomicExpression( AtomicExpressionContext ctx ){
+		if(ctx.IDENTIFIER() != null){
+			return new AstVariable(ctx.getText());
+		}
+		if(ctx.NUMERIC_LITERAL() != null){
+			return new AstLiteral(NumericalEvaluator.parse(ctx.getText()));
+		}
+		if(ctx.LPAREN() != null){
+			return this.visitLogicalOrExpression(ctx.logicalOrExpression());
```

Comment:
You could use `visit` here, then it will be dispatched automatically based on the context. Otherwise, you will need to make periodic adjustments here as the grammar changes.